src/backend/base/langflow/base/tools/flow_tool.py

```diff
@@ -99,20 +99,38 @@ def build_tweaks_dict(self, args, kwargs):
         kwargs = self.validate_inputs(args_names=args_names, args=args, kwargs=kwargs)
         return {arg["component_name"]: kwargs[arg["arg_name"]] for arg in args_names}
 
+    def build_inputs_dict(self, args, kwargs) -> list[dict[str, Any]]:
+        inputs = self.inputs
+        components = [input_.display_name for input_ in inputs]
+        if len(args) == len(inputs):
+            return [
+                {"components": components, "type": input_.vertex_type.lower(), "input_value": value}
+                for input_, value in zip(inputs, args, strict=False)
+            ]
+
+        return [
+            {
+                "components": components,
+                "type": input_.vertex_type.lower(),
+                "input_value": kwargs[input_.display_name.lower().strip().replace(" ", "_")],
+            }
+            for input_ in inputs
+        ]
+
     async def _arun(
         self,
         *args: Any,
         **kwargs: Any,
     ) -> str:
         """Use the tool asynchronously."""
-        tweaks = self.build_tweaks_dict(args, kwargs)
+        inputs = self.build_inputs_dict(args, kwargs)
         try:
             run_id = self.graph.run_id if hasattr(self, "graph") and self.graph else None
         except Exception:  # noqa: BLE001
             logger.opt(exception=True).warning("Failed to set run_id")
             run_id = None
         run_outputs = await run_flow(
-            tweaks={key: {"input_value": value} for key, value in tweaks.items()},
+            inputs=inputs,
```

Comment:
You have to be mindful here that the values passed here are not necessarily inputs to the flow. That is why we built tweaks.

Reply:
Thanks. Updated